tests/unit/tests_standard_lib/test_event_ingestors/test_file_monitor_ingestor.py

```diff
@@ -0,0 +1,199 @@
+required_config = {
+    "uf_host": "localhost",
+    "uf_port": "8888",
+    "uf_username": "admin",
+    "uf_password": "secret",
+}
+sample_event = namedtuple("SampleEvent", ["event", "metadata"], defaults=[None, None])
+
+
+def test_ingest(mock_object):
+    mock_object(f"{file_name}.sleep")
+    events = ["event_1", "event_2", "event_3"]
+    with patch.object(
+        FileMonitorEventIngestor, "create_output_conf"
+    ) as create_output_conf_mock, patch.object(
+        FileMonitorEventIngestor, "create_event_file"
+    ) as create_event_file_mock, patch.object(
+        FileMonitorEventIngestor, "create_inputs_stanza"
+    ) as create_inputs_stanza_mock:
+        fmei = FileMonitorEventIngestor(required_config)
+        fmei.ingest(events, 20)
+        create_output_conf_mock.assert_called_once_with()
+        create_event_file_mock.assert_has_calls([call(event) for event in events])
+        create_inputs_stanza_mock.assert_has_calls([call(event) for event in events])
+        assert (
+            create_event_file_mock.call_count
+            == create_inputs_stanza_mock.call_count
+            == len(events)
+        )
+
+
+def test_create_output_conf(requests_mock):
+    requests_mock.post("https://localhost:8888/services/data/outputs/tcp/group")
+    fmei = FileMonitorEventIngestor(required_config)
+    fmei.create_output_conf()
+    sent_requests = [
+        (unquote(str(req)), unquote(req.text)) for req in requests_mock.request_history
+    ]
+    assert sent_requests == [
+        (
+            "POST https://localhost:8888/services/data/outputs/tcp/group",
+            "name=uf_monitor&servers=splunk:9997",
+        )
+    ]
+
+
+@pytest.mark.parametrize(
+    "response, message",
+    [
+        (
+            {"text": "Not Found", "status_code": 404},
+            "Unable to create stanza in outputs.conf\nStatus code: 404 \nReason: None \ntext:Not Found",
+        ),
+        (
+            {"exc": ConnectionError("test connection error")},
+            "Unable to connect to Universal forwarder, test connection error",
+        ),
+    ],
+)
+def test_create_output_conf_bad_request(requests_mock, caplog, response, message):
+    requests_mock.post(
+        "https://localhost:8888/services/data/outputs/tcp/group", **response
+    )
+    fmei = FileMonitorEventIngestor(required_config)
+    fmei.create_output_conf()
+    assert message in caplog.messages
+
+
+def test_create_event_file(open_mock):
+    with patch.object(
+        FileMonitorEventIngestor, "get_file_path", return_value="fake_file"
+    ):
+        fmei = FileMonitorEventIngestor(required_config)
+        fmei.create_event_file(
+            sample_event(
+                event="CREATE EVENT",
+                metadata={"host": "localhost", "source": "sys.log"},
+            )
+        )
+        open_mock.assert_has_calls(
+            [
+                call("fake_file", "w+"),
+                call().__enter__(),
```

Comment:
I think checking `__enter__` and `__exit__` calls ca be removed

Reply:
as discussed I will leave it as it is